src/include/wiredtiger.in

```diff
@@ -446,6 +478,38 @@ struct __wt_cursor {
 	int __F(insert)(WT_CURSOR *cursor);
 
 	/*!
+	 * Modify an existing record.
+	 *
+	 * Both the key and value must be set and the record must already exist;
+	 * the record will be updated.
+	 *
+	 * Modification structures are applied in order, and later modifications
+	 * can update earlier modifications.
+	 *
+	 * The modify method is only supported on raw byte arrays accessed using
+	 * a WT_ITEM structure, that is, a format type of \c u.
+	 *
+	 * @snippet ex_all.c Modify an existing record
+	 *
+	 * On success, the cursor ends positioned at the modified record; to
+	 * minimize cursor resources, the WT_CURSOR::reset method should be
+	 * called as soon as the cursor no longer needs that position.
+	 *
+	 * The maximum length of a single column stored in a table is not fixed
+	 * (as it partially depends on the underlying file configuration), but
+	 * is always a small number of bytes less than 4GB.
+	 *
+	 * @param cursor the cursor handle
+	 * @param entries an array of modification data structures
+	 * @param nentries the number of modification data structures
+	 * @errors
+	 * In particular, if \c in_memory is configured for the database and
+	 * the modify requires more than the configured cache size to complete,
+	 * ::WT_CACHE_FULL is returned.
+	 */
+	int __F(modify)(WT_CURSOR *cursor, WT_MODIFY *entries, int nentries);
```

Comment:
I've been thinking about this - and I think the interface would be more consistent if it were a change to ``WT_CURSOR::set_value()`` rather than this which is a corollary of ``WT_CURSOR::update``, but adds a data setting interface. Using the ``WT_CURSOR::set_value`` approach means we get the benefit of being able to use the functionality for tables with more complex schema definitions. The complexity with my suggested approach is that we currently assume that ``u`` are ``WT_ITEM`` in the code, and we'd need to make cases aware that it could now be a ``WT_MODIFY``, and we'd possibly need another level of nesting in the ``WT_MODIFY`` structure that encapsulates the nentries field (which is obviously clunky or you would have done that).

@michaelcahill @keithbostic what do you think?